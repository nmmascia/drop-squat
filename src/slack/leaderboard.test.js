const leaderboard = require('./leaderboard');

test('should return the correct tally sections', () => {
  expect(leaderboard.countHeader()).toMatchSnapshot();
});

test('should return the correct header sections', () => {
  expect(leaderboard.boardHeader(new Date('March 26 1988'))).toMatchSnapshot();
});

test('should return the correct section for count under 3', () => {
  expect(leaderboard.userCount({ userId: 'nick', count: 2 })).toMatchSnapshot();
});

test('should return the correct section for count above 3', () => {
  expect(leaderboard.userCount({ userId: 'nick', count: 4 })).toMatchSnapshot();
});

test('should return the correct section for baotw', () => {
  expect(leaderboard.userCount({ userId: 'nick', count: 2, baotw: true })).toMatchSnapshot();
  expect(leaderboard.userCount({ userId: 'nick', count: 4, baotw: true })).toMatchSnapshot();
});
