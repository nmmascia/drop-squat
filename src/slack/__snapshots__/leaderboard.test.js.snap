// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return the correct header sections 1`] = `
Object {
  "text": Object {
    "text": "*Leaderboard for Mar 26th, 1988*",
    "type": "mrkdwn",
  },
  "type": "section",
}
`;

exports[`should return the correct section for baotw 1`] = `
Object {
  "fields": Array [
    Object {
      "text": ":zombie::trophy: <@nick>",
      "type": "mrkdwn",
    },
    Object {
      "text": "2",
      "type": "plain_text",
    },
  ],
  "type": "section",
}
`;

exports[`should return the correct section for baotw 2`] = `
Object {
  "fields": Array [
    Object {
      "text": ":trophy: <@nick>",
      "type": "mrkdwn",
    },
    Object {
      "text": "4",
      "type": "plain_text",
    },
  ],
  "type": "section",
}
`;

exports[`should return the correct section for count above 3 1`] = `
Object {
  "fields": Array [
    Object {
      "text": "<@nick>",
      "type": "mrkdwn",
    },
    Object {
      "text": "4",
      "type": "plain_text",
    },
  ],
  "type": "section",
}
`;

exports[`should return the correct section for count under 3 1`] = `
Object {
  "fields": Array [
    Object {
      "text": ":zombie:<@nick>",
      "type": "mrkdwn",
    },
    Object {
      "text": "2",
      "type": "plain_text",
    },
  ],
  "type": "section",
}
`;

exports[`should return the correct tally sections 1`] = `
Object {
  "fields": Array [
    Object {
      "text": "*Squatter*",
      "type": "mrkdwn",
    },
    Object {
      "text": "*Workouts*",
      "type": "mrkdwn",
    },
  ],
  "type": "section",
}
`;
